thrust/cmake/thrust-config.cmake

```diff
@@ -0,0 +1,322 @@
+    endif()
+  endif()
+endif()
+
+if (TARGET Thrust::OMP)
+  _thrust_setup_host_device_targets(OMP)
+endif()
+
+#
+# Standardize version info
+#
+
+if (NOT THRUST_VERSION)
+  set(THRUST_VERSION ${${CMAKE_FIND_PACKAGE_NAME}_VERSION})
+  set(THRUST_VERSION_MAJOR ${${CMAKE_FIND_PACKAGE_NAME}_VERSION_MAJOR})
+  set(THRUST_VERSION_MINOR ${${CMAKE_FIND_PACKAGE_NAME}_VERSION_MINOR})
+  set(THRUST_VERSION_PATCH ${${CMAKE_FIND_PACKAGE_NAME}_VERSION_PATCH})
+  set(THRUST_VERSION_TWEAK ${${CMAKE_FIND_PACKAGE_NAME}_VERSION_TWEAK})
+  set(THRUST_VERSION_COUNT ${${CMAKE_FIND_PACKAGE_NAME}_VERSION_COUNT})
+endif()
+
+if (TARGET Thrust::CUDA AND NOT THRUST_VERSION VERSION_EQUAL CUB_VERSION)
+  message(FATAL_ERROR
+    "The version of CUB found by CMake is not compatible with this release of Thrust. "
+    "CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. "
+    "Set the CMake option THRUST_IGNORE_CUB_VERSION_CHECK=ON before calling find_package(Thrust) to ignore this check."
+  )
+endif()
+
+#
+# Helper functions:
+#
+
+# thrust_mark_options_advanced() will hide all THRUST_* options from CMake GUIs
+# This is to future-proof projects that want to hide any future options that may
+# be added.
+function(thrust_mark_options_advanced)
+  mark_as_advanced(
+    THRUST_HOST_SYSTEM
+    THRUST_DEVICE_SYSTEM
+  )
+endfunction()
+
+# thrust_debug_targets() will print details of all provided targets.
+function(thrust_debug_targets)
+  function(_thrust_debug_target target version)
+    if (NOT TARGET ${target})
+      return()
+    endif()
+
+    message(STATUS "Target: ${target} (${version})")
+
+    function(_thrust_print_prop_if_set target prop)
+      get_target_property(value ${target} ${prop})
+      if (value)
+        message(STATUS "  ${prop}: ${value}")
+      endif()
+    endfunction()
+
+    _thrust_print_prop_if_set(${target} ALIASED_TARGET)
+    _thrust_print_prop_if_set(${target} INTERFACE_COMPILE_DEFINITIONS)
+    _thrust_print_prop_if_set(${target} INTERFACE_COMPILE_FEATURES)
+    _thrust_print_prop_if_set(${target} INTERFACE_COMPILE_OPTIONS)
+    _thrust_print_prop_if_set(${target} INTERFACE_INCLUDE_DIRECTORIES)
+    _thrust_print_prop_if_set(${target} INTERFACE_LINK_DEPENDS)
+    _thrust_print_prop_if_set(${target} INTERFACE_LINK_DIRECTORIES)
+    _thrust_print_prop_if_set(${target} INTERFACE_LINK_LIBRARIES)
+    _thrust_print_prop_if_set(${target} INTERFACE_LINK_OPTIONS)
+    _thrust_print_prop_if_set(${target} INTERFACE_SYSTEM_INCLUDE_DIRECTORIES)
+
+  endfunction()
+
+  function(_thrust_debug_backend_targets backend version)
+    _thrust_debug_target(Thrust::${backend} "${version}")
+    _thrust_debug_target(Thrust::${backend}::Host "${version}")
+    _thrust_debug_target(Thrust::${backend}::Device "${version}")
+  endfunction()
+
+  _thrust_debug_target(Thrust::Thrust "v${THRUST_VERSION}")
+  _thrust_debug_target(Thrust::Thrust::Impl "v${THRUST_VERSION}")
```

Comment:
I think we should do "Thrust X.Y.Z" not "Thrust vX.Y.Z", just for consistency.

Reply:
Fixed locally.